File: Stack/LinkedList/Java/Stack.java
class Stack<T>{

  private Element head;
  private int size;

  public Stack(){
    this.head = null;
    this.size = 0;
  }

  public void push(T data){
    Element newE;
    if(head == null){
      newE = new Element(data, null);
    }else{
      newE = new Element(data, head);
    }
    head = newE;
    this.size++;
  }

  public T pop(){
      if(head == null){
        return null;
      }
      T ret = this.head.data;
      this.head = this.head.prev;
      this.size--;
      return ret;
  }

  public int returnSize(){
    return this.size;
  }

  public boolean isEmpty(){
    return this.size == 0;
  }


  private class Element{

    T data;
    Element prev;

    public Element(T data, Element prev){
      this.data = data;
      this.prev = prev;
    }
  }
}

File: Stack/LinkedList/C/Stack.c
#include "Stack.h"
#include <stdlib.h>

Element* createElement(int data, Element *prev){
  Element *e = (Element*) malloc(sizeof(Element));
  e->data = data;
  e->prev = prev;
  return e;
}

Stack* createStack(){
  Stack *stack = (Stack*) malloc(sizeof(Stack));
  stack->head = NULL;
  stack->size = 0;
}

void push(Stack *stack, int data){
  Element *newE;
  if(stack->head == NULL){
    newE = createElement(data, NULL);
  }else{
    newE = createElement(data, stack->head);
  }
  stack->head = newE;
  stack->size++;
}

int pop(Stack *stack){
  if(stack->head == NULL){
    return -1;
  }
  int ret = stack->head->data;
  stack->head = stack->head->prev;
  stack->size--;
  return ret;
}

int returnSize(Stack *stack){
    return stack->size;
}

int isEmpty(Stack *stack){
  return stack->size == 0;
}
